### PR TITLE
fix(trading): prevent crash when all wallets are ejected

### LINKED ISSUE #27431: App crashes when all wallets are ejected from wallet switcher during trading
**Describe the bug**
Ejecting all wallets from the wallet switcher while in the trading tab causes Trezor Suite to crash with a fatal error.

**Info:**
 - Suite version: desktop 26.5.0 (b4cd9298c64b46f6336a74ea839e40428dd549b6)
 - Browser: Trezor Suite 26.5.0
 - OS: Linux x86_64
 - Screen: 3440x1440
 - Device: Trezor T3W1 2.11.2 universal (revision c2025931d489dc0bccd8daad84f876bdd4677e5f)
 - Transport: BridgeTransport 3.2.1, BluetoothTransport

**Steps to reproduce**
1. Go to the Trading tab
2. Eject all wallets from the wallet switcher
3. Observe app crash (see screenshot)

**Expected behavior**
App should not crash. Show error or disable trading when there are no wallets.

**Screenshots**
error screen after ejecting all wallets in trading

https://github.com/user-attachments/assets/db1ad942-dd82-4cce-8627-e97c333e9761

**Additional context**
Error message: Cannot read properties of undefined (reading 'key')

### PR DESCRIPTION
**Pull request by sherpaPSX**
## Description

- Stop rendering trading form content when the wallet switcher has no visible accounts.
- Show the existing discovery empty state in trading instead of crashing with a fatal error.
- Keep the trading navigation visible while the empty state is shown.
- Add a TradingLayout regression test for both populated and empty-account states.

## Notes for QA

- Open Trading, eject all wallets from the wallet switcher, and verify the tab shows the empty discovery state instead of a fatal error.
- Re-enable or reconnect a wallet, open Trading again, and verify the regular trading content renders.

## Related Issue

Resolve #27431

## Screenshots:

N/A

<details>
<summary><h3>🤖 LLM Test Recommendations</h3></summary>

**Summary:** The change modifies TradingLayout.tsx, a shared layout component that wraps all trading views (Buy, Sell, Swap). The second changed file is a unit test for TradingLayout itself, which has no E2E coverage. Since TradingLayout is a broad component used by 17 tests, the strategy focuses on one high-priority navigation test that validates all three trading form types render correctly, plus representative medium-priority tests covering complete Buy, Sell, and Swap flows to ensure the layout doesn't break any trading path. The risk is moderate — layout changes can affect rendering and visibility of child components across all trading views.

<details><summary><b>Changed files (2)</b></summary>

- `packages/suite/src/views/wallet/trading/common/TradingLayout/TradingLayout.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingLayout/__tests__/TradingLayout.test.tsx`

</details>

**Recommended tests (6)**

<details><summary>🔴 <b>High priority (1)</b></summary>

- `suite/e2e/tests/trading/navigation.test.ts` — TradingLayout is the shared layout component wrapping all trading views (Buy, Sell, Swap). This navigation test exercises opening Buy, Sell, and Swap forms from multiple entry points and verifying form rendering — directly validating that the TradingLayout renders correctly and doesn't break form visibility across all trading types.

</details>

<details><summary>🟡 <b>Medium priority (4)</b></summary>

- `suite/e2e/tests/trading/buy-bitcoin.test.ts` — This test exercises the full Buy flow including form rendering, quote display, offer confirmation, and transaction detail — all rendered within the TradingLayout. It validates that the layout correctly hosts the buy form and its various states (quotes, confirmation, detail).
- `suite/e2e/tests/trading/sell-solana.test.ts` — This test covers a complete Sell flow with quotes, confirmation, device prompt, transaction status, and offer comparison — all rendered within TradingLayout. It validates the layout works correctly for the sell trading path.
- `suite/e2e/tests/trading/swap-coins.test.ts` — This test exercises the full swap flow including form, quotes, device confirmation, toast notifications, and transaction status progression — all hosted within TradingLayout. It validates layout integrity for the swap trading path.
- `suite/e2e/tests/trading/buy-negative.test.ts` — Tests error states in the buy form (max/min limit errors, empty quotes, disabled buttons) which render inside TradingLayout. A layout change could affect error message visibility or button state rendering.

</details>

<details><summary>🟢 <b>Low priority (1)</b></summary>

- `suite/e2e/tests/trading/sell-inputs.test.ts` — Tests sell form input validation (decimals, insufficient funds, percentage buttons) rendered within TradingLayout. Less likely to be affected by a layout-level change but could surface layout-related rendering issues in form inputs.

</details>

⚠️ **Changes with no test coverage (1)**
- `packages/suite/src/views/wallet/trading/common/TradingLayout/__tests__/TradingLayout.test.tsx`

_Updated: 2026-06-01T07:08:57.221Z_

</details>

## 🔍 Currents Test Results

🔍 **Suite desktop test results:** [View in Currents](https://app.currents.dev/projects/4ytF0E/runs?branch=fix/27431-trading-wallet-eject-crash&lookback=7)

🔍 **Suite web test results:** [View in Currents](https://app.currents.dev/projects/Og0NOQ/runs?branch=fix/27431-trading-wallet-eject-crash&lookback=7)

## 🔒 Quarantined E2E Tests

<details><summary>Trezor Suite (web) — 5 test(s)</summary>

| Test | Type |
|------|------|
| Account metadata > dropbox provider | 🤖 auto |
| Quarantine test: "Onboarding - create wallet,Success (basic)" | 🙋 manual |
| Quarantine test: "Database migration,Db migration between: release/22.5/web => develop/web" | 🙋 manual |
| Quarantine test: "Multiple sessions,Session overtaken by another" | 🙋 manual |
| Quarantine test: "Recovery T2T1 - dry run,Recovery after partial recovery" | 🙋 manual |

</details>

_Updated: 2026-06-02T10:21:35.714Z • 5 test(s) total_

<details><summary>Trezor Suite (desktop) — 6 test(s)</summary>

| Test | Type |
|------|------|
| Cardano > Basic cardano walkthrough | 🤖 auto |
| Account types suite > Add-account-types-non-BTC-coins | 🤖 auto |
| Passphrase with cardano > verify cardano address behind passphrase | 🤖 auto |
| Public Keys > Check ada XPUB | 🤖 auto |
| Onboarding - create wallet > Success (basic) | 🤖 auto |
| Quarantine test: "Multiple sessions,Session overtaken by another" | 🙋 manual |

</details>

_Updated: 2026-06-01T07:12:23.155Z • 6 test(s) total_

## 🌐 Preview deployments

🌐 **Suite Web preview:** [https://dev.suite.sldev.cz/suite-web/fix/27431-trading-wallet-eject-crash/web/](https://dev.suite.sldev.cz/suite-web/fix/27431-trading-wallet-eject-crash/web/)

### CODE REVIEW

File: packages/suite/src/views/wallet/trading/common/TradingLayout/TradingLayout.tsx [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
 
 export const TradingLayout = ({ children }: PropsWithChildren) => {
     const routeName = useSelector(selectRouteName);
+    const hasVisibleAccounts = useSelector(state => selectVisibleDeviceAccounts(state).length > 0);
 
     return (
         <Column data-testid="@trading" gap={24}>
             <TradingLayoutNavigation route={routeName} />
-            {children}
+            {hasVisibleAccounts ? (
```
Comment (vytick) [inline packages/suite/src/views/wallet/trading/common/TradingLayout/TradingLayout.tsx]: what if there is no connected account?

Comment (vytick) [reply packages/suite/src/views/wallet/trading/common/TradingLayout/TradingLayout.tsx]: we probably want to show that screen - `connect your device` content

**Comment by vytick** [comment]
@coderabbitai review

**Review by vytick — APPROVED** [review]
